src/measure_theory/bochner_integration.lean

```diff
@@ -1055,42 +1055,51 @@ lemma integral_non_measurable (h : ¬ measurable f) : integral f = 0 :=
 integral_undef $ not_and_of_not_left _ h
 
 variables (α β)
-@[simp] lemma integral_zero : integral (0 : α → β) = 0 :=
+@[simp] lemma integral_zero : integral (λ(a:α), (0:β)) = 0 :=
```

Comment:
Does this still trigger as `simp`-lemma after your proposed change? Same elsewhere.

Reply:
I tested it.
```lean
import data.real.basic algebra.pi_instances

constant integral : (ℝ → ℝ) → ℝ

@[simp] lemma simp1 : integral (0 : ℝ → ℝ) = 0 := sorry
lemma foo : integral (λa:ℝ, (0:ℝ)) = 0 := by simp -- fails
lemma foo' : integral (0 : ℝ → ℝ) = 0 := by simp -- succeeds
lemma bar : integral (λa:ℝ, (0:ℝ)) = 0 := by rw simp1 -- fails
lemma bar' : integral (0 : ℝ → ℝ) = 0 := by rw simp1 -- succeeds
```
```lean
import data.real.basic algebra.pi_instances

constant integral : (ℝ → ℝ) → ℝ

@[simp] lemma simp2 : integral (λa:ℝ, (0:ℝ)) = 0 := sorry

lemma foo : integral (λa:ℝ, (0:ℝ)) = 0 := by simp -- succeeds
lemma foo' : integral (0 : ℝ → ℝ) = 0 := by simp -- fails
lemma bar : integral (λa:ℝ, (0:ℝ)) = 0 := by rw simp2 -- succeeds
lemma bar' : integral (0 : ℝ → ℝ) = 0 := by rw simp2 -- fails
```
It seems that one of them will fail in any case. 

Johannes told me to make this change at the fomm meeting. I think the reason is that, when an expression is long and difficult, it is more likely to be a lambda term and that is when we need `simp` and `rw` the most.

Reply:
And this is more in line with the `integration.lean` file, 
```lean
@[simp] lemma lintegral_zero : (∫⁻ a:α, 0) = 0 :=
```
where `(∫⁻ a:α, 0) ` is really `lintegral (λa:α 0)`.

Reply:
I would suggest that we have both versions as simp-lemma. So maybe one of the two styles can get a `'` at the end of the name. @urkud Didn't you have something similar recently?

Reply:
Maybe I can use the binder notation, then people don’t have a choice?

Reply:
I've never tried that myself, but it sounds reasonable.

Reply:
I had something similar with derivatives: sometimes you want to simplify `deriv cos` without an argument (e.g., in `deriv (deriv cos)`), and sometimes you want to rewrite `-sin x` back to `deriv cos x`.
We'll probably need something similar once we'll have FTC and a file with a table of antiderivatives.
As for `f+g` etc, derivatives always use `λx, f x + g x`.